pkg/lib/config/configsource/database.go

```diff
@@ -221,13 +222,19 @@ func (d *Database) resolveAppIDByDomain(ctx context.Context, r *http.Request) (s
 	return appID, nil
 }
 
-func (d *Database) ResolveContext(ctx context.Context, appID string) (*config.AppContext, error) {
+func (d *Database) ResolveContext(ctx context.Context, appID string) (context.Context, *config.AppContext, error) {
 	value, _ := d.appMap.LoadOrStore(appID, &dbApp{
 		appID: appID,
 		load:  &sync.Once{},
 	})
 	app := value.(*dbApp)
-	return app.Load(ctx, d)
+	appCtx, err := app.Load(ctx, d)
+	if err != nil {
+		return nil, nil, err
```

Comment:
In case of error, we should return the original context. It is safer to return the original context so that the caller can write

```golang
defer func() {
  if err != nil {
    // ctx is not nil here. So doSomethingWithCtx will not be broken.
    doSomethingWithCtx(ctx, ...)
  }
}()
ctx, appCtx, err = ResolveContext()
```